erpnext/schools/doctype/assessment_result_tool/assessment_result_tool.js

```diff
@@ -7,17 +7,19 @@ cur_frm.add_fetch("assessment_plan", "student_batch", "student_batch");
 frappe.ui.form.on('Assessment Result Tool', {
 
 	onload: function(frm) {
+		frm.trigger("refresh");
+	},
+
+	refresh: function(frm) {
 		if (frappe.route_options) {
 			frm.doc.assessment_plan = frappe.route_options.assessment_plan;
+			frm.trigger("assessment_plan");
```

Comment:
and why you are forcefully triggering this on every refresh?

As I can see, it resets the table. Hence, if you go to other tabs after entering some data and come back to the same tab, it will call refresh and eventually will clear the data.